contracts/RelayHub.sol

```diff
@@ -224,9 +225,14 @@ contract RelayHub is IRelayHub {
 
         bool forwarderSuccess;
         bytes memory relayedCallReturnValue;
-        //use succ as relay call success variable
-        (forwarderSuccess, destinationCallSuccess, relayedCallReturnValue) = Eip712Library
-            .execute(relayRequest, signature);
+        if(collectorContract == address(0)) {
+            (forwarderSuccess, destinationCallSuccess, relayedCallReturnValue) = Eip712Library
+                .execute(relayRequest, signature);
+        } else {
+            (forwarderSuccess, destinationCallSuccess, relayedCallReturnValue) = Eip712Library
+                .execute(relayRequest, signature, collectorContract);
+        }
```

Comment:
unnecessary check, as `execute(relayRequest, signature)` will call the method with `address(0)`, which is the collectorContract in this case anyway.